pkg/storage/gcp/saver.go

```diff
@@ -0,0 +1,93 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/gobuffalo/envy"
+	"google.golang.org/api/option"
+)
+
+// Storage implements the (github.com/gomods/pkg/storage).Saver interface
+type Storage struct {
+	bucket *storage.BucketHandle
+}
+
+// New returns a new Storage instance authenticated using the provided
+// ClientOptions. The bucket name to be used will be loaded from the
+// environment variable ATHENS_GCP_BUCKET_NAME.
+//
+// The ClientOptions should provide permissions sufficient to read, write and
+// delete objects in google cloud storage for your project.
+func New(ctx context.Context, cred option.ClientOption) (*Storage, error) {
+	client, err := storage.NewClient(ctx, cred)
```

Comment:
Since the client is created here, and it might not be returned below (in an error case, like if `envy.MustGet` returns an error), what do you think about returning a `func()` from this function that calls `client.Close()` if appropriate?

we may have talked about this before, but I don't remember exactly what we decided on, so if I'm repeating something that we already decided on, forgive me 😄

Reply:
We can add this, and the Close method in the Storage struct. It is not required to be closed on program termination as per the gcp storage library docs, but if we do it would take a small amount off the GC.

Reply:
ah yea, now I remember. if `gcp.New(...)` is only called once (it looks like it is?) then it seems ok to leave as-is